### PR TITLE
FLOW-10403:Nylas endpoints to upload attachments, download attachment…

### PR DESCRIPTION
**Pull request by arturpakula**
… metadata and delete uploaded attachments

### CODE REVIEW

File: src/service/NylasService/NylasService.ts [hunk 1]
```diff
@@ -65,6 +65,49 @@ export class NylasService extends APIClient {
         });
     }
 
+    /**
+     * Returns an attachment metadata using the nylas api
+     * @param emailAccount the email to be sending from
+     * @param attachmentId
+     */
+    async getAttachmentMetadata(emailAccount: string, attachmentId: string): Promise<AxiosResponse> {
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:73]: Endpoints which are fetching things have to be prefixed with `fetch` 
```suggestion
    async fetchAttachmentMetadata(emailAccount: string, attachmentId: string): Promise<AxiosResponse> {
```

File: src/service/NylasService/NylasService.ts [hunk 2]
```diff
@@ -65,6 +65,49 @@ export class NylasService extends APIClient {
...
+        return await this.invokeApi(`/nylas/files/${attachmentId}`, 'GET', '', {
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:74]: You should always use `invokeApiWithErrorHandling`

File: src/service/NylasService/NylasService.ts [hunk 3]
```diff
@@ -65,6 +65,49 @@ export class NylasService extends APIClient {
...
+    /**
+     * Uploads an attachment using the nylas api
+     * @param emailAccount the email to be sending from
+     * @param file attachment
+     * @param cancelToken
+     */
+    async uploadAttachment(emailAccount: string, file: Blob, cancelToken?: CancelToken): Promise<AxiosResponse> {
+        const formData = new FormData();
+        formData.append('file', file);
+        return await this.invokeApi('/nylas/files', 'POST', formData, {
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:90]: You should always use `invokeApiWithErrorHandling`

File: src/service/NylasService/NylasService.ts [hunk 4]
```diff
@@ -65,6 +65,49 @@ export class NylasService extends APIClient {
...
+        });
+    }
+
+    /**
+     * Removes an attachment using the nylas api
+     * @param emailAccount the email to be sending from
+     * @param attachmentId
+     */
+    async removeAttachment(emailAccount: string, attachmentId: string): Promise<AxiosResponse> {
+        return await this.invokeApi(`/nylas/files/${attachmentId}`, 'DELETE', '', {
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:104]: You should always use `invokeApiWithErrorHandling`

File: src/service/NylasService/NylasService.ts (hunk 2 above)
Comment (ffoliversaternus) [inline src/service/NylasService/NylasService.ts:74]: Better use invokeApiWithErrorhandling

File: src/service/NylasService/NylasService.ts (hunk 1 above)
Comment (ffoliversaternus) [inline src/service/NylasService/NylasService.ts:73]: The response body here is probably an object. So you should use an interface for that.

**Review by ffjanhoeck — CHANGES_REQUESTED** [review]

**Review by ffoliversaternus — CHANGES_REQUESTED** [review]
Add Typing and error handling for all requests where necessary

File: src/service/NylasService/NylasService.ts [hunk 5]
```diff
@@ -57,14 +57,57 @@ export class NylasService extends APIClient {
...
         });
     }
 
+    /**
+     * Returns an attachment metadata using the nylas api
+     * @param emailAccount the email to be sending from
+     * @param attachmentId
+     */
+    async fetchAttachmentMetadata(emailAccount: string, attachmentId: string): Promise<ApiResponse<any>> {
+        return await this.invokeApiWithErrorHandling(`/nylas/files/${attachmentId}`, 'GET', '', {
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:74]:
```suggestion
        return await this.invokeApiWithErrorHandling(`/nylas/files/${attachmentId}`, 'GET', undefined, {
```

File: src/service/NylasService/NylasService.ts [hunk 6]
```diff
@@ -57,14 +57,57 @@ export class NylasService extends APIClient {
...
+        });
+    }
+
+    /**
+     * Removes an attachment using the nylas api
+     * @param emailAccount the email to be sending from
+     * @param attachmentId
+     */
+    async removeAttachment(emailAccount: string, attachmentId: string): Promise<ApiResponse<any>> {
+        return await this.invokeApiWithErrorHandling(`/nylas/files/${attachmentId}`, 'DELETE', '', {
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:104]:
```suggestion
        return await this.invokeApiWithErrorHandling(`/nylas/files/${attachmentId}`, 'DELETE', undefined, {
```

**Review by ffjanhoeck — APPROVED** [review]

**Pull request merged**